frontend/src/app/modules/user/user.component.ts

```diff
@@ -12,34 +14,57 @@ import { TabRoute } from '@shared/models/tabs/tab-route';
 })
 export class UserComponent implements OnInit {
   currentUser: User = {} as User;
+  githubClick = false;
+  googleClick = false;
 
   tabRoutes: TabRoute[] = [
     { name: 'Profile', route: '' },
     { name: 'Project settings', route: '' },
     { name: 'New settings', route: '' },
   ];
 
-  constructor(private route: ActivatedRoute, private cropper: ModalCropperService, private userService: UserService) { }
+  constructor(
+    private route: ActivatedRoute,
+    private cropper: ModalCropperService,
+    private userService: UserService,
+    private fbr: FirebaseSignInService) { }
 
   ngOnInit(): void {
-    this.route.data.subscribe( data => this.currentUser = data.user);
-    this.userService.userLogoUrl.subscribe( url => {
+    this.route.data.subscribe(data => this.currentUser = data.user);
+    this.userService.userLogoUrl.subscribe(url => {
       this.currentUser.avatarUrl = url;
     });
-    this.userService.userLogoUserName.subscribe( userName => {
+    this.userService.userLogoUserName.subscribe(userName => {
       this.currentUser.username = userName;
     });
   }
 
-  async open(){
+  async open() {
     const file = await this.cropper.open();
-    if (file){
-      console.log('we have cropped ' + typeof(file));
+    if (file) {
+      console.log('we have cropped ' + typeof (file));
       // now we can use it for saving image logic
     }
-    else{
+    else {
       console.log('Image didn`t change');
     }
   }
 
+  linkWithGithub() {
+    this.fbr.linkWithGithub();
+  }
+
+  linkWithGoogle() {
+    this.fbr.linkWithGoogle();
+  }
+
+  isProviderAdded(provider: Providers) {
+    for (const item of this.currentUser.userSocialNetworks) {
+      if (item.socialNetworkId - 1 === Number(provider)) {
```

Comment:
You can change it Array.some(...). But still what do you do with item.socialNetworkId - 1

Reply:
I take it into account